src/queue_internal.h

```diff
@@ -0,0 +1,20 @@
+#include <stddef.h>
```

Comment:
This file can be removed & the `queue_element_type` can be defined in the c file.